File: app/src/main/java/com/danilomendes/figurines/data/remote/CompanyRemoteDataSource.java
package com.danilomendes.figurines.data.remote;

import com.danilomendes.figurines.data.remote.api.CompanyService;

import javax.inject.Inject;

/**
 * Created by danilo on 08-11-2017.
 */
public class CompanyRemoteDataSource {
    private final NetworkManager mNetworkManager;


    @Inject
    public CompanyRemoteDataSource(NetworkManager networkManager) {
        this.mNetworkManager = networkManager;
    }

    public CompanyService getCompaniesService() {
        return mNetworkManager.getRetrofit().create(CompanyService.class);
    }

    public boolean isNetworkAvailable() {
        return mNetworkManager.isNetworkAvailable();
    }
}

File: app/src/main/java/com/danilomendes/figurines/data/remote/CompanyRemoteDataSource.kt
package com.danilomendes.figurines.data.remote

import com.danilomendes.figurines.data.remote.api.CompanyService
import javax.inject.Inject

/**
 * Created by danilo on 08-12-2017.
 */
class CompanyRemoteDataSource @Inject constructor(
        private val networkManager: NetworkManager) {

    fun getCompaniesService(): CompanyService {
        return networkManager.retrofit.create(CompanyService::class.java)
    }

    fun isNetworkAvailable(): Boolean {
        return networkManager.isNetworkAvailable()
    }
}